### PR TITLE
raidboss: Additional M/R2S triggers

### PR DESCRIPTION
**Pull request by wexxlee**
Finally getting around to PRing some additional triggers:
* Revises the 'Stored: ' partner/spread call to an `infoText` and suppresses it for the first two occurrences (since Tempting Twist/Honey Beeline happen immediately after).
* Adds an `infoText` with the subsequent positioning + partner/spread at the beginning of Tempting Twist and Honey Beeline, followed by an explicit `alertText` with that info when the first part of the mechanic resolves.
* Lists some possible TODOs for beat1 & beat2.
* Adds defamation/in/towers calls during beat3.  (NB: I had this mostly worked up before #349, but if people prefer that approach + some added handling for defamations, I can remove it from this PR before merging.)
* Adds callouts for "curtain call" (aka Beeloved Venom).  On this, I added a 'Marge w/ X` alert for the active player, as well as info alerts for other merges that should happen not involving the active player.  On the latter, I could see some disagreement about whether that info is useful or not, so happy to adjust/remove that if folks don't like.

(NB2: This PR doesn't address/replace anything in #339 or #343 -- I had in my notes to do things similarly to those, but since there are already PRs, I think it makes sense to work with those.)

This looks fine in vod/RE, but probably could use a proofread and some in-game testing.  I may be able to test a little later tonight, but would be great if others can poke at it too.

### CODE REVIEW

**Comment by JLGarber** [comment]
I will not have time to fully review this tonight, but on a quick read through it all looks fine. Nothing jumped out at me as being obviously incorrect or poorly-handled. I'll need to have a think about this vs 349's handling on the Defamation sequence.

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (niaka3dayo) [inline ui/raidboss/data/07-dt/raid/r2s.ts:43]: [MAY] It would be more natural and easier to understand in Japanese to write `自分に巨大な爆発` (`jibun ni kyodai na bakuhatsu`) rather than `自分の巨大な爆発` (`jibun no kyodai na bakuhatsu`).

```suggestion
    ja: '自分に巨大な爆発',
```

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (valarnin) [inline ui/raidboss/data/07-dt/raid/r2s.ts:525]: Pure subjective suggestion:

Here (and below for other), 9s might be pushing the timing a bit too tight?

The exact timings are:
12s - initial debuffs
16.5s - Time between initial debuffs and first raidwide
7s - Duration of magic vuln from merging with other player

This only leaves a 6.5 second window for players to react to the callout and move to merge with the other player, and have the server actually count them as merging.

It might make sense to have this be 10s instead of 9s? Possibly only for the first merge?

**Review by valarnin — COMMENTED** [review]
I don't really have time tonight to do a deep dive on this either, but a quick skim through showed nothing glaring.

> Adds callouts for "curtain call" (aka Beeloved Venom). On this, I added a 'Marge w/ X` alert for the active player, as well as info alerts for other merges that should happen not involving the active player. On the latter, I could see some disagreement about whether that info is useful or not, so happy to adjust/remove that if folks don't like.

I think that the "other" triggers are useful both as a healer and as the person doing callouts for my group. Making the "self" trigger `alertText` and the "other" trigger `infoText` should be sufficient enough to distinguish them and players always have the option of disabling the trigger in raidboss config if it's too noisy for them personally?

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r2s.ts:525]: > Pure subjective suggestion:
> 
> Here (and below for other), 9s might be pushing the timing a bit too tight?
> 
> The exact timings are: 12s - initial debuffs 16.5s - Time between initial debuffs and first raidwide 7s - Duration of magic vuln from merging with other player
> 
> This only leaves a 6.5 second window for players to react to the callout and move to merge with the other player, and have the server actually count them as merging.
> 
> It might make sense to have this be 10s instead of 9s? Possibly only for the first merge?

Yeah, I definitely think this could be adjusted.  I seem to remember some concern about merging too early (maybe just allowing time to heal up from the raidwide? Can't recall if the raidwide also inflicts a vuln) -- you probably have a better perspective on that as a healer than I do though.

I wasn't able to test this in-game tonight -- hopefully tomorrow at some point -- so let me play with the timing a little and see what feels comfy.

(As an aside, this prompted me to finally get around to #358, so if that lands, I could see it being applied here or added in a later PR to help give a better timeframe to players.)

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r2s.ts:43]: > [MAY] It would be more natural and easier to understand in Japanese to write `自分に巨大な爆発` (`jibun ni kyodai na bakuhatsu`) rather than `自分の巨大な爆発` (`jibun no kyodai na bakuhatsu`).

I copied this from the defamation in P9S - should it be updated there too?  Or is this change something more unique to this particular mechanic?

Comment (niaka3dayo) [reply ui/raidboss/data/07-dt/raid/r2s.ts:43]: > > [MAY] It would be more natural and easier to understand in Japanese to write `自分に巨大な爆発` (`jibun ni kyodai na bakuhatsu`) rather than `自分の巨大な爆発` (`jibun no kyodai na bakuhatsu`).
> 
> I copied this from the defamation in P9S - should it be updated there too? Or is this change something more unique to this particular mechanic?

I see. This isn't unique to this particular mechanic. Rather, I think we should modify 'the defamation' in P95 similarly to use 'jibun ni' (自分に) to keep the expression consistent across the board. Would you like me to create a small PR for that change?

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (valarnin) [reply ui/raidboss/data/07-dt/raid/r2s.ts:525]: The raidwide doesn't inflict a vuln, so it's safe to merge as soon as party health/shields/mits allow it after the raidwide.

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (JLGarber) [reply ui/raidboss/data/07-dt/raid/r2s.ts:43]: Is this common to all Defamation-style mechanics? [Alexander 12 Savage features a similar mechanic](https://github.com/OverlayPlugin/cactbot/blob/70eb959f8a1a907a886743f3d1e6577cb31668f7/ui/raidboss/data/03-hw/raid/a12s.ts#L206), and so does[ Alexander Ultimate](https://github.com/OverlayPlugin/cactbot/blob/70eb959f8a1a907a886743f3d1e6577cb31668f7/ui/raidboss/data/05-shb/ultimate/the_epic_of_alexander.ts#L2017), in [two separate places](https://github.com/OverlayPlugin/cactbot/blob/70eb959f8a1a907a886743f3d1e6577cb31668f7/ui/raidboss/data/05-shb/ultimate/the_epic_of_alexander.ts#L2157). (There are other places within the codebase, but these are just the first ones that came to mind.) If it's something that should be changed across the board like that, it might be best to add this to `outputs.ts` and then manually update the combination mechanics like the second iteration I linked on Alexander Ultimate.

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (JLGarber) [reply ui/raidboss/data/07-dt/raid/r2s.ts:525]: I haven't been using triggers on floors 1/2 for various reasons, but as a shot caller, I agree that a 10s warning would be better here. People are sometimes *really* slow to move in when it's their turn, so "merge soon" a little earlier seems better to me, as opposed to "merge now".

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (niaka3dayo) [reply ui/raidboss/data/07-dt/raid/r2s.ts:43]: > Is this common to all Defamation-style mechanics? [Alexander 12 Savage features a similar mechanic](https://github.com/OverlayPlugin/cactbot/blob/70eb959f8a1a907a886743f3d1e6577cb31668f7/ui/raidboss/data/03-hw/raid/a12s.ts#L206), and so does[ Alexander Ultimate](https://github.com/OverlayPlugin/cactbot/blob/70eb959f8a1a907a886743f3d1e6577cb31668f7/ui/raidboss/data/05-shb/ultimate/the_epic_of_alexander.ts#L2017), in [two separate places](https://github.com/OverlayPlugin/cactbot/blob/70eb959f8a1a907a886743f3d1e6577cb31668f7/ui/raidboss/data/05-shb/ultimate/the_epic_of_alexander.ts#L2157). (There are other places within the codebase, but these are just the first ones that came to mind.) If it's something that should be changed across the board like that, it might be best to add this to `outputs.ts` and then manually update the combination mechanics like the second iteration I linked on Alexander Ultimate.

I completely agree with your suggestion. This mechanic appears quite frequently across various battle contents in FF14, so managing it centrally in `outputs.ts` would indeed be an effective and efficient approach.

Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r2s.ts:43]: I'll update the PR to drop this in `outputs.ts`, and I'll separately PR the change for the other places its used (so as not to clutter this PR).

**Comment by wexxlee** [comment]
(Rebased after the other r2s PRs were merged, and added a suppress to the beat2 spread/towers trigger that was missing from #343.)

This should be ready for merge.  I've tested in game a couple times, and everything looks good.  On the timing of the curtain call stuff, I opted for 10s as suggested.  10s definitely felt comfy for the first merge -- for the 2nd-4th merge, a 10s call comes out right as the prior raidwide damage is being applied, so a player with itchy trigger fingers could see that and merge before being healed up.  But delaying it by 1 second doesn't really add much healing margin, and any later than that feels rushed (plus using different delays based on merge order feels like overkill), so I think a flat 10s, plus using "Merge Soon" instead of just "Merge", is the best of the options.

**Review by JLGarber — APPROVED** [review]
Sorry about the delay, Life In General has been a thing this week. This all looks good. I'll let you push the button on the merge in case any last-minute changes jump out at you.

**Comment by jacob-keller** [comment]
I'm going to pull this and manually test it in my raid tonight!

**Comment by jacob-keller** [comment]
I have a couple of thoughts:

- I don't like dropping the info callouts for the first two venoms, it makes it much harder on me as a caller. There is still enough of a delay I think we should keep them
- Somewhere along the way we started getting a double callout of the SIDES
- The defamation/towers callouts worked great during beat3
- The debuff cleansing during the final mechanic worked ok, but I had trouble actually doing callout with it. I think thats on me though.

I think this is overall positive, but I do think I would like to keep the info venom callouts.

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (jacob-keller) [inline ui/raidboss/data/07-dt/raid/r2s.ts:307]: this block seems like a duplicate that doesn't need to be added.

**Review by jacob-keller — CHANGES_REQUESTED** [review]

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (jacob-keller) [reply ui/raidboss/data/07-dt/raid/r2s.ts:307]: That, or remove the original report.

Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r2s.ts:307]: > this block seems like a duplicate that doesn't need to be added.

> That, or remove the original report.

I'm not sure what you mean.  A duplicate of what?  What's the "original report" that you're referring to?

**Comment by wexxlee** [comment]
> * I don't like dropping the info callouts for the first two venoms, it makes it much harder on me as a caller. There is still enough of a delay I think we should keep them

I don't think we usually do "stored" or "later"-type calls when the mechanic happens immediately after, but if you think it's helpful to have these for the first two, I don't have a major issue with keeping the info callouts there.

> * Somewhere along the way we started getting a double callout of the SIDES

Not seeing that in the handful of pulls from this week or in prior logs running through RE.  Could you provide a split/anonymized log with the duplicated callout?

**Comment by jacob-keller** [comment]
> > * I don't like dropping the info callouts for the first two venoms, it makes it much harder on me as a caller. There is still enough of a delay I think we should keep them
> 
> I don't think we usually do "stored" or "later"-type calls when the mechanic happens immediately after, but if you think it's helpful to have these for the first two, I don't have a major issue with keeping the info callouts there.
> 

I would like to keep the info callouts. Maybe without saying "stored"? It helps me because I like to call out "it will be partners/spreads" and its easier to do that with an info callout. Otherwise you eaither delay saying its partners or spread until after the sides or donut appear. Its not a huge deal and I would just port it back in on my instance.


> > * Somewhere along the way we started getting a double callout of the SIDES
> 
> Not seeing that in the handful of pulls from this week or in prior logs running through RE. Could you provide a split/anonymized log with the duplicated callout?

I think this is a merge conflict with some other changes I had actually. You can ignore that.

File: ui/raidboss/data/07-dt/raid/r2s.ts
Comment (jacob-keller) [inline ui/raidboss/data/07-dt/raid/r2s.ts:309]: I think what happened is that I had a merge conflict with  something on my tree that made me accidentally keep the old implementation in.

**Pull request merged**
